org.librarysimplified.r2.vanilla/src/main/java/org/librarysimplified/r2/vanilla/internal/SR2Controller.kt

```diff
@@ -411,12 +415,31 @@ internal class SR2Controller private constructor(
   private fun executeCommandBookmarkDelete(
     apiCommand: SR2Command.BookmarkDelete
   ): ListenableFuture<*> {
-    val newBookmarks = this.bookmarks.toMutableList()
-    val removed = newBookmarks.remove(apiCommand.bookmark)
-    if (removed) {
-      this.bookmarks = newBookmarks.toList()
-      this.eventSubject.onNext(SR2BookmarkDeleted(apiCommand.bookmark))
+    this.bookmarks = this.bookmarks.map { bookmark ->
+      bookmark.copy(
+        isBeingDeleted = bookmark == apiCommand.bookmark
+      )
     }
+    this.eventSubject.onNext(
+      SR2BookmarkTryToDelete(
+        bookmark = apiCommand.bookmark,
+        onDeleteOperationFinished = { wasDeleted ->
+          if (wasDeleted) {
+            val newBookmarks = this.bookmarks.toMutableList()
+            newBookmarks.remove(apiCommand.bookmark)
+            this.bookmarks = newBookmarks.toList()
+            this.eventSubject.onNext(SR2BookmarkDeleted(apiCommand.bookmark))
+          } else {
+            this.bookmarks = this.bookmarks.map { bookmark ->
+              bookmark.copy(
+                isBeingDeleted = false
+              )
```

Comment:
This sets `isBeingDeleted` of _all_ bookmarks to false, but do we want to just set it for `apiCommand.bookmark`? I guess the question is, can there be more than one bookmark deletion in progress at the same time? If so, then if one fails, this would change the state of all the ones in progress.

Reply:
Yes, you're right. This is supposed to change just one bookmark and not all of them. Nice catch, thank you!